auth.js

```diff
@@ -0,0 +1,137 @@
+    emailElementParent.append(emailErrorMsgElement);
+  } else {
+    email.classList.remove("error");
+    const emailExistingError = emailElementParent.querySelector(".error_msg");
+    if (emailExistingError) emailExistingError.remove();
+  }
+
+  submitButtonVisible();
+};
+
+const checkPasswordData = (e) => {
+  const passwordElementParent = e.target.parentNode;
+  const passwordErrorMessegeEmpty = "비밀번호를 입력해주세요.";
+  const passwordErrorMessegeFormat = "비밀번호를 8자 이상 입력해주세요.";
+
+  let passwordErrorMsgElement = document.createElement("p");
+  passwordErrorMsgElement.className = "error_msg";
+
+  const passwordExistingErrorMsg = passwordElementParent.querySelector(".error_msg");
+  if (passwordExistingErrorMsg) {
+    passwordExistingErrorMsg.remove();
+  }
+
+  if (password.value === "") {
+    password.classList.add("error");
+    passwordErrorMsgElement.textContent = passwordErrorMessegeEmpty;
+    passwordElementParent.append(passwordErrorMsgElement);
+  } else if (password.value.length < 8) {
+    password.classList.add("error");
+    passwordErrorMsgElement.textContent = passwordErrorMessegeFormat;
+    passwordElementParent.append(passwordErrorMsgElement);
+  } else {
+    password.classList.remove("error");
+    const passwordExistingError = passwordElementParent.querySelector(".error_msg");
+    if (passwordExistingError) passwordExistingError.remove();
+  }
+
+  submitButtonVisible();
+};
+
+const checkPasswordConfirmData = (e) => {
+  const confirmElementParent = e.target.parentNode;
+  const confirmErrorMessegeEmpty = "비밀번호를 입력해주세요.";
+  const confirmErrorMessege = "비밀번호가 일치하지 않습니다..";
+
+  let confirmErrorMsgElement = document.createElement("p");
+  confirmErrorMsgElement.className = "error_msg";
+
+  const confirmExistingErrorMsg = confirmElementParent.querySelector(".error_msg");
+  if (confirmExistingErrorMsg) {
+    confirmExistingErrorMsg.remove();
+  }
+
+  if (passwordConfirm.value === "") {
+    passwordConfirm.classList.add("error");
+    confirmErrorMsgElement.textContent = confirmErrorMessegeEmpty;
+    confirmElementParent.append(confirmErrorMsgElement);
+  } else if (password.value !== passwordConfirm.value) {
+    passwordConfirm.classList.add("error");
+    confirmErrorMsgElement.textContent = confirmErrorMessege;
+    confirmElementParent.append(confirmErrorMsgElement);
+  } else {
+    passwordConfirm.classList.remove("error");
+    const confirmExistingError = confirmElementParent.querySelector(".error_msg");
+    if (confirmExistingError) confirmExistingError.remove();
+  }
+
+  submitButtonVisible();
+};
+
+function submitButtonVisible() {
+  const emailCheck = email ? emailRegex.test(email.value) : true;
+  const passwordCheck = password ? password.value.length >= 8 : true;
+  const passwordConfirmCheck = passwordConfirm ? password.value === passwordConfirm.value && passwordConfirm.value !== "" : true;
+
+  if (loginButton) {
+    const loginCehck = emailCheck && passwordCheck;
```

Comment:
❗️ 수정요청
오타가 있네요~

```suggestion
    const loginCheck = emailCheck && passwordCheck;
```

참고로 loginCheck라는 변수명은 어떤 값을 의미하는지 알기 어려우므로 isValid, isLoginFormValid 같이 타입과 값을 추론할 수 있는 이름을 추천드립니다.